### PR TITLE
snowcow sol

### PR DESCRIPTION
**Pull request by SansPapyrus683**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making the changes.

### CODE REVIEW

**Review by ryanchou-dev — APPROVED** [review]
circles at the chevron

File: solutions/usaco-973.mdx [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+			if (l_closest != painted.begin()
+					&& end[sb] <= end[(--l_closest)->second]) {
+				continue;
+			}
+			// remove all children of the current snowball that are in the map
+			while (true) {
+				auto r_next = painted.upper_bound(start[sb]);
+				if (r_next == painted.end() || end[sb] <= end[r_next->second]) {
+					break;
+				}
```
[the remark this answers is not in the document]
Comment (SansPapyrus683) [reply solutions/usaco-973.mdx:189]: not sure how to do that without making a mess of variables

File: solutions/usaco-973.mdx [hunk 2]
```diff
@@ -0,0 +1,371 @@
...
+using std::endl;
+using std::vector;
+
+//BeginCodeSnip{Binary Indexed Tree}
+class BITree {
+	private:
+		vector<long long> bit;
+		int size;
+	public:
+		BITree(int size) : size(size), bit(size) { }
```
Comment (dongliu0426) [inline solutions/usaco-973.mdx:91]: huh how does this not RTE? BIT is 1-indexed right?

Comment (SansPapyrus683) [reply solutions/usaco-973.mdx:91]: trust me, it works

**Review by dongliu0426 — APPROVED** [review]

**Pull request merged**